node/node_syncing.go

```diff
@@ -61,10 +62,9 @@ SyncingLoop:
 				node.stateSync = syncing.CreateStateSync(node.SelfPeer.IP, node.SelfPeer.Port, node.GetSyncID())
 			}
 			if node.stateSync.GetActivePeerNumber() == 0 {
-				if node.stateSync.CreateSyncConfig(getPeers()) {
-					node.stateSync.MakeConnectionToPeers()
-				} else {
-					utils.GetLogInstance().Debug("[SYNC] no active peers, continue SyncingLoop")
+				peers := getPeers()
+				if err := node.stateSync.CreateSyncConfig(peers); err != nil {
+					ctxerror.Log15(utils.GetLogInstance().Debug, err)
```

Comment:
what's this?

Reply:
ctxerror is the structured (context-aware) error support added in response to your comment about #675.  It enables a function to return an error with named context objects, much like how log15 loggers take message along with named context objects.

`Log15` takes a log15-style error logging function and an error instance, and simply logs the error message if the error instance is a genetic error type, or logs the message with context parameters if the error instance is a ctxerror.  I agree that the syntax is a bit backward/functional-programming-y/feels like a Russian reversal joke; suggestions are welcome on how to improve this.

Reply:
why only here use this Log15 function?

Reply:
I did this specific error refactoring because it was blocking a clean reduction of layering interdependencies (and also, returning only a true/false boolean indicator to the caller is not an idiomatic Go pattern).  Feel free to use `ctxerror` elsewhere you see fit though.